source/utils.rst

```diff
@@ -0,0 +1,173 @@
+.. role:: raw-html(raw)
+   :format: html
+.. role:: bash(code)
+   :language: bash
+
+#########
+Utilities
+#########
+
+Utilities for handling feedstocks at `nsls-ii-forge <https://github.com/nsls-ii-forge>`_.
+
+Here is a list of current functioning utilties:
+
+* check-results
+  
+  * Check the various parameters of a generated conda package
+
+* all-feedstocks
+  
+  * Clone/List all available feedstock repositories
+
+============
+Installation
+============
+
+At the command line
+
+.. code-block:: bash
+
+    $ pip install nsls2forge-utils
+
+=====================
+GitHub Authentication
+=====================
+
+Authentication is done through :bash:`~/.netrc` and :bash:`~/.conda-smithy/github.token`.
+
+GitHub Access Token
+===================
+
+First, create a `personal access token <https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token>`_ for GitHub.
+
+netrc Setup
+===========
+
+This may already be setup if you automatically authenticate GitHub using git.
+
+If not, create a file and open it for editing:
+
+.. code-block:: bash
+
+    $ vi ~/.netrc
+
+Place the following in that file with your GitHub username and access token.
+
+.. code-block:: bash
+
+    machine github.com
+    login [USERNAME]
+    password [ACCESS_TOKEN]
+
+To save the file and quit hit ESC, type :bash:`wq` and hit ENTER.
```

Comment:
I think we also need instructions to make this file owner-readable only. Something like the following will work:
```bash
$ chmod 600 ~/.netrc
```
This is to secure the sensitive information in that plain-text file. Same should be done with the `~/.conda-smithy/` dir/files. The following command should remove "group" and "others" access bits:
```bash
$ chmod -Rv go-rwx ~/.conda-smithy
```
(but it may already be covered by the previous documentation about setting up the system).